cmake/Helpers.cmake

```diff
@@ -515,9 +515,9 @@ FUNCTION(CONFIGURE_APPLE_SPECIFICS)
 	FIND_LIBRARY(COCOA_LIBRARY_PATH Cocoa)
 	FIND_LIBRARY(COREFOUNDATION_LIBRARY CoreFoundation)
 	# for objective-c++
-	STRING(APPEND CMAKE_CXX_FLAGS " -DTARGET_OS_MAC")
+	SET(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} " -DTARGET_OS_MAC")
 	# for pure objective-c
-	STRING(APPEND CMAKE_C_FLAGS " -x objective-c -DTARGET_OS_MAC")
+	SET(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} " -x objective-c -DTARGET_OS_MAC")
```

Comment:
How are these two things different?
Anyway, since I expect this to fail, the cmake way to add definitions is https://cmake.org/cmake/help/latest/command/add_compile_definitions.html#command:add_compile_definitions